CDCgov/prime-reportstream
# Fetch Refactor: Hook up RSErrorBoundary and Suspense to ValueSetsIndex and ValueSetsDetail

This PR removes error handling from react-query hooks and puts the responsibility on the component rendering system. Using helper functions, we can now wrap components with error UI and spinner UI helpers (RSErrorBoundary and Suspense). 

See docs for information.

Notable changes:
- React Query hook tests no longer test for errors, error handling integrations tests are the component's responsibility
- Added response interfaces

Test Steps:
1. Ensure the UI behaves as intended by loading the Value Sets Index and Value Sets Detail pages.
2. Ensure tests pass by running yarn test

## Changes
- *Include a comprehensive list of changes in this PR*
- *(For web UI changes) Include screenshots/video of changes*

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #6522 

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
